compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -855,7 +857,8 @@ class Semantic {
       case vdef : ValDef =>
         // local val definition
         // TODO: support explicit @cold annotation for local definitions
-        eval(vdef.rhs, thisV, klass).ensureHot("Local definitions may only hold initialized values", vdef)
+        eval(vdef.rhs, thisV, klass, true)
```

Comment:
I've added cacheResult = true here to cache the result for rhs, but I've found that in several test cases, we evaluate rhs again when accessing local variable after. Is this the right way to cache the result?

Reply:
Use `cache = true` instead of `true` for better readability.